jboss/container/wildfly/launch-config/config/added/launch/launch.sh

```diff
@@ -39,18 +50,258 @@ function configure_server() {
+
+  # In case a script is generated by extensions and needs to be run against server started in admin mode. NB: server will be reloaded in normal mode or asked to be restarted.
+  handleExtensions
+  if [ $? -ne 0 ]; then
+    log_info "Restarting the server"
+    ${cmd} &
+    pid=$!
+  fi
+  wait $pid 2>/dev/null
+  wait $pid 2>/dev/null
+}
+
+# Functions to handle CLI execution during boot
+
+function configure_boot_script() {
+  local script="$1"
+  # remove any empty line
+  if [ -f "${script}" ]; then
+    sed -i '/^$/d' $script
+  fi
+
+  if [ -f "$JBOSS_HOME/extensions/postconfigure.sh" ] || [ -f "$JBOSS_HOME/extensions/delayedpostconfigure.sh" ]; then
+    CLI_BOOT_HANDLE_EXTENSIONS="true"
+    # Corner case, no main script although extensions.
+    if ! [ -f "${script}" ]; then
+      # we need an existing empty script file for the server to generate the marker file advertising that it is ready to receive extensions script CLI operations.
+      echo "" > "${script}"
+    fi
+  fi
+
+  if [ -s "${script}" ] || [ -n "${CLI_BOOT_HANDLE_EXTENSIONS}" ]; then
+    local t=$(date +%s)
+    CLI_BOOT_RELOAD_MARKER_DIR=/tmp/cli-boot-reload-marker-${t}
+    mkdir -p "${CLI_BOOT_RELOAD_MARKER_DIR}"
+    CLI_EXECUTION_OPTS="--start-mode=admin-only -Dorg.wildfly.internal.cli.boot.hook.script=${script} \
+      -Dorg.wildfly.internal.cli.boot.hook.marker.dir=${CLI_BOOT_RELOAD_MARKER_DIR} \
+      -Dorg.wildfly.internal.cli.boot.hook.script.properties=${CLI_SCRIPT_PROPERTY_FILE} \
+      -Dorg.wildfly.internal.cli.boot.hook.script.output.file=${CLI_SCRIPT_OUTPUT_FILE} \
+      -Dorg.wildfly.internal.cli.boot.hook.script.error.file=${CONFIG_ERROR_FILE} \
+      -Dorg.wildfly.internal.cli.boot.hook.script.warn.file=${CONFIG_WARNING_FILE}"
+
+    if [ -n "${CLI_BOOT_HANDLE_EXTENSIONS}" ]; then
+      CLI_EXECUTION_OPTS="${CLI_EXECUTION_OPTS} -Dorg.wildfly.internal.cli.boot.hook.reload.skip=true"
+      log_info "Extensions detected, server will be reloaded from remote CLI script."
+    fi
+    log_info "Server started in admin mode, CLI script executed during server boot."
+
+    if [ "${SCRIPT_DEBUG}" = "true" ] ; then
+      log_info "Server options: ${CLI_EXECUTION_OPTS}" 
+      log_info "CLI Script generated by launch: ${script}"
+      log_info "CLI Script property file: ${CLI_SCRIPT_PROPERTY_FILE}"
+      log_info "CLI Script error file: ${CONFIG_ERROR_FILE}"
+      log_info "CLI Script output file: ${CLI_SCRIPT_OUTPUT_FILE}"
+    fi
+  else
+    if [ "${SCRIPT_DEBUG}" = "true" ]; then
+      log_info "================= CLI files debug ================="
+      log_info "No CLI commands were found in ${script}"
+    fi 
+  fi
+}
+
+function handleExtensions {
+  if [ -n "${CLI_BOOT_HANDLE_EXTENSIONS}" ]; then
+    local marker_file=${CLI_BOOT_RELOAD_MARKER_DIR}/wf-cli-invoker-result
+    wait_marker "${marker_file}"
+    if [ $? -ne 0 ]; then
+      log_error "Error, server never advertised configuration done. Can't proceed with extensions."
```

Comment:
minor gripe but "extension" as a very specific meaning in WildFly ecosystem. As I understand it, in this case "extensions" mean "CLI configuration extension" or something like that. It might be better to rephrase it with a more descriptive sentence

Reply:
I suggest: "Can't proceed with custom extensions script."